src/main/java/io/thinkit/edc/client/connector/services/ApplicationObservability.java

```diff
@@ -25,15 +27,8 @@ public ApplicationObservability(String url, HttpClient httpClient, UnaryOperator
         this.interceptor = interceptor;
     }
 
-    public Result<HealthStatus> checkHealth() {
+    Result<HealthStatus> getResponse(HttpResponse<InputStream> response) {
```

Comment:
these internal methods should have `private` visibility` and put below the public ones